learning_journal_db/tests.py

```diff
@@ -0,0 +1,185 @@
+import pytest
+import transaction
+from pyramid import testing
+from .models import (
+    get_engine,
+    get_session_factory,
+    get_tm_session,
+)
+from .models.mymodel import MyModel
+from .models.meta import Base
+
+
+@pytest.fixture(scope='session')
+def sqlengine(request):
+    config = testing.setUp(settings={
+        'sqlalchemy.url': 'sqlite:///:memory:'
+        })
+    config.include('.models')
+    settings = config.get_settings()
+    engine = get_engine(settings)
+    Base.metadata.create_all(engine)
+
+    def teardown():
+        testing.tearDown()
+        transaction.abort()
+        Base.metadata.drop_all(engine)
+
+    request.addfinalizer(teardown)
+    return engine
+
+
+@pytest.fixture(scope='function')
+def new_session(sqlengine, request):
+    session_factory = get_session_factory(sqlengine)
+    dbsession = get_tm_session(session_factory, transaction.manager)
+
+    def teardown():
+        transaction.abort()
+
+    request.addfinalizer(teardown)
+    return dbsession
+
+# Testing the models.
+
+
+def test_model_gets_added(new_session):
+    """Test the creation of the new model."""
+    assert len(new_session.query(MyModel).all()) == 0
+    model = MyModel(title="test_day", body="test_body")
+    new_session.add(model)
+    new_session.flush()
+    assert len(new_session.query(MyModel).all()) == 1
+
+
+def test_model_gets_added_2(new_session):
+    """Test the creation of 2 new models in the row."""
+    assert len(new_session.query(MyModel).all()) == 0
+    model = MyModel(title="test_day", body="test_body")
+    new_session.add(model)
+    new_session.flush()
+    model = MyModel(title="test_day2", body="test_body2")
+    new_session.add(model)
+    new_session.flush()
+    assert len(new_session.query(MyModel).all()) == 2
```

Comment:
While this is good, it might help you to create a helper function that puts multiple model instances into the database. Perhaps something that takes a number as an argument and adds that number of model instances to your database. That way you wouldn't have to write this sequence over and over again.